dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -1180,30 +1080,96 @@ public Object putIfAbsent(String key, Object value) {
     public boolean remove(Object key, Object value) {
         Integer index = KEYMAP.getOrDefault(key, KEY_NOT_FOUND);
         if (index != KEY_NOT_FOUND) {
-            if (propertyValues[index] == value) {
-                propertyValues[index] = NOT_FOUND;
-                return true;
-            } else {
+            if (propertyValues[index] != NOT_FOUND) {
+                if (propertyValues[index] == null) {
+                    if (value == null) {
+                        propertyValues[index] = NOT_FOUND;
+                        return true;
+                    } else {
+                        return false;
+                    }
+                } else if (propertyValues[index].equals(value)){
+                    propertyValues[index] = NOT_FOUND;
+                    return true;
+                }
                 return false;
             }
+            return false;
         }
         return super.remove(key, value);
     }
     @Override
     public Object compute(String key, BiFunction<? super String, ? super Object, ? extends Object> remappingFunction) {
-        throw new UnsupportedOperationException();
+        Integer index = KEYMAP.getOrDefault(key, KEY_NOT_FOUND);
+        if (index != KEY_NOT_FOUND) {
+            Object newValue = remappingFunction.apply(key, propertyValues[index] == NOT_FOUND ? null : propertyValues[index]);
+            if (propertyValues[index] != NOT_FOUND) {
```

Comment:
I think this if / else can be simplified to:

propertyValues[index] = newValue == null ? NOT_FOUND : newValue;

Reply:
If remappingFunction == null this method should throw NPE.